### PR TITLE
Replace temperature display by query frequency

### PR DESCRIPTION
**Pull request by DL6ER**
# What does this implement/fix?

See title.

image

**Related issue or feature (if applicable):** N/A

**Pull request in [docs](https://github.com/pi-hole/docs) with documentation (if applicable):** N/A

---
**By submitting this pull request, I confirm the following:** 

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against. 
2. I have commented my proposed changes within the code.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

## Checklist:

- [x] The code change is tested and works locally.
- [x] I based my code and PRs against the repositories `developmental` branch.
- [x] I [signed off](https://docs.pi-hole.net/guides/github/how-to-signoff/) all commits. Pi-hole enforces the [DCO](https://docs.pi-hole.net/guides/github/dco/) for all contributions
- [x] I [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) all my commits. Pi-hole requires signatures to verify authorship
- [x] I have read the above and my PR is ready for review.

### CODE REVIEW

**Comment by yubiuser** [comment]
Personally, I don't like 'Queries per minute'. I think it is confusion when the 'scale' changes depending on the number of events. Additionally, fractions of q/s doesn't make sense to me. I suggest to just round it to integer and show '<1 q/s' for really low values.

**Comment by rdwebdesign** [comment]
>  I don't like 'Queries per minute'.

Actually, initially I though this was a good idea, specially in systems with a very low query frequency, but changing `q/s` to `q/m` is too subtle (only one letter is changed) and users will not notice the unit change. 

If we decide to keep 'Queries per minute' for cases with small frequencies, I still think we should use `q/min`.

**Another suggestion:**

> I think it is confusion when the 'scale' changes depending on the number of events.

We can keep the change between 'Queries per second' and 'Queries per minute', but we could also change the color (to blue or something else) when the unit changes to "per minute" to make the change more visible.

**Comment by DL6ER** [comment]
With your two suggestions
- always show `q/s` (no conversion to per-minute), and
- show `< 1 q/s` for smaller frequencies
we can go ahead and simply hard-code the `< 1 q/s` string and don't bother with actually computing the real number on most systems.

You will have to see more than `24*60*60 = 86,400` per 24 hours to get the value `1 q/s` and already need double that number (`172,800` queries in 24 hrs) to get a `2 q/s`.

---

So should it always be `per-min` ?

**Comment by rdwebdesign** [comment]
> So should it always be `per-min` ?

I think this is a good idea. 

Maybe we could reverse the strategy and use a different unit only in extreme cases, if there are too many queries per minute ([like this](https://discourse.pi-hole.net/t/pi-hole-large-scale/72261)), to avoid a text "overflow", but this is very uncommon.

**Comment by yubiuser** [comment]
> You will have to see more than 24*60*60 = 86,400 per 24 hours to get the value 1 q/s and already need double that number (172,800 queries in 24 hrs) to get a 2 q/s.

I'm not sure if I can follow your logic here. Sure the calculation is correct, but the info shown is not intended to show the 24 hour average, but the 'current' query stream. 
But I agree, maybe it's better to show queries/min by default and only above a threshold for high load switch to queries/sec

**Comment by DL6ER** [comment]
> I'm not sure if I can follow your logic here. Sure the calculation is correct, but the info shown is not intended to show the 24 hour average, but the 'current' query stream.

However, I have seen in many screenshots that the query "load" is rather even, e.g., on my own Pi-hole, I see
image

so I'll barely see 1 q/s while for most it way always be `< 1 q/s` - okay, maybe +/- for a few seconds at peak times every day

**Comment by yubiuser** [comment]
Your distribution is quite distinct from mine...



__

So where are we going from here? 
Default to queries/min and switch to queries/sec above e.g. 100 q/sec (for really large installations or loop detection)?

**Comment by PromoFaux** [comment]
wildcard idea - make it configurable... either /s or /min dependent on user preference.

**Comment by DL6ER** [comment]
More configurable means more likeliness to break and a lot more maintenance. I adopted the idea suggested above.

image

File: scripts/pi-hole/js/footer.js [hunk 1]
```diff
@@ -226,15 +225,45 @@ function initCheckboxRadioStyle() {
...
+  if (freq > 100) {
+    freq /= 60;
+    unit = "q/s";
+    title = "Queries per second";
+  }
+
+  // Determine number of fraction digits based on the frequency
+  // - 0 fraction digits for frequencies > 10
+  // - 1 fraction digit for frequencies between 1 and 10
+  // - 2 fraction digits for frequencies < 1
```
Comment (yubiuser) [inline scripts/pi-hole/js/footer.js:249]: I still think we should only go with full integer values and "<1" only.

Comment (rdwebdesign) [reply scripts/pi-hole/js/footer.js:249]: I think showing decimal places only for small numbers (`< 10`) is fine. 
We already show decimal places for the CPU and MEM.

Comment (DL6ER) [reply scripts/pi-hole/js/footer.js:249]: Are we making any progress on this one? I still think low numbers deserve some decimal places as to not artificially loose information when it is there - even on low query count

Comment (rdwebdesign) [reply scripts/pi-hole/js/footer.js:249]: I agree with DL6ER. We could show the decimal places for small numbers.

Comment (yubiuser) [reply scripts/pi-hole/js/footer.js:249]: Ok, then proceed with the current implementation. Time and users will tell if we need to change :-)

**Review by yubiuser — APPROVED** [review]

**Pull request merged**